### ISSUE TITLE
Blockchain Implementation

### ISSUE THREAD

**Issue by rulonsw** [rulonsw/ChainReact#6]
This epic encompasses the work required to add blockchain-backed posts to the system.

**Comment by aussieninja** [comment]
With this milestone, I think it's important to note that only users that have hooked up their app account to Steem should be able to post to the Steem Blockchain.  Eventually I'd this this app (if possible) to allow users to post their content to Instagram, Twitter & Steem by selecting checkboxes and clicking post.

This app doesn't need to allow users to convert Steem in their wallets, etc (we can just link to https://steemitwallet.com/) but the blockchain integration that I think is important is:

- Posting to the Steem blockchain if you've hooked up your account.
- Users who have hooked up Steem should then vote on other content using their Steem voting power and resource credits.
- Anyone can make comments, but users hooked up to Steem will make comments using their Steem account while users not hooked up to Steem will still comment to Steem (if the original post is on the blockchain) but the comment can be posted by the application account.

Basically users be able to use this app fully with or without a Steem account... those that don't have an account will just earn points for their content that can be exchanged for, say, Amazon giftcards, while people who have hooked up their Steem account (which should be able to be done after account creation) will get points and steem upvotes.
